collect_app/src/main/java/org/odk/collect/android/widgets/WidgetFactory.java

```diff
@@ -194,11 +193,8 @@ public static QuestionWidget createWidgetFromPrompt(FormEntryPrompt prompt,
             case Constants.CONTROL_AUDIO_CAPTURE:
                 RecordingRequester recordingRequester;
 
-                Optional<TreeElement> audioQuality = prompt.getBindAttributes().stream().filter(attr -> {
-                    return attr.getName().equals("audio-quality");
-                }).findAny();
-
-                if (audioQuality.isPresent() && audioQuality.get().getAttributeValue().equals("external")) {
+                Optional<String> audioQuality = FormEntryPromptUtils.getAttributeValue(prompt, "audio-quality");
+                if (audioQuality.isPresent() && audioQuality.get().equals("external")) {
```

Comment:
This feels like a very `null` structure if you see what I mean. Not really much benefit to an `Optional` if you're not chaining, for example.